### PR TITLE
Part 2 of Colors 2 Integration

### PR DESCRIPTION
**Pull request by edjamesmsft**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Binary change:

| File | Before | After | Delta |
|------|--------|-------|-------|
| libFluentUI.a | 28,830,392  | 28,832,200 | 1808 |

Moved the demo color provider to using ColorProviding2, implementing the ColorProviding2 protocol. Updated demos that were incorrectly colored.

### Verification

Built and ran Simulator, comparing before and after against figma.

| Before                                       | After                                      | Green                                          |
|----------------------------------------------|--|------------------------------------------|
| HUDLightBefore | HUDLightAfter| |
| HUDDarkBefore | HUDDarkAfter | |
| SegmentedControlLightBefore | SegmentedControlLightAfter | SegmentedControlGreen |
| SegmentedControlDarkBefore | SegmentedControlDarkAfter | |
| TVCLightBefore | TVCLightAfter | TVCGreen |
| TVCDarkBefore | TVCDarkAfter | |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1496)

### CODE REVIEW

File: ios/FluentUI/SegmentedControl/SegmentedControl.swift [hunk 1]
```diff
@@ -425,7 +425,6 @@ open class SegmentedControl: UIView, TokenizedControlInternal {
         for (index, button) in buttons.enumerated() {
             button.updateTokenizedValues()
             button.setTitleColor(contentColor, for: .normal)
```
Comment (huwilkes) [inline ios/FluentUI/SegmentedControl/SegmentedControl.swift:427]: If you're removing tintColor, you might also want to remove setTitleColor? They're both the old way.

File: ios/FluentUI/Table View/TableViewCell.swift [hunk 2]
```diff
@@ -44,7 +44,7 @@ public enum TableViewCellAccessoryType: Int {
         case .detailButton:
             return UIColor(dynamicColor: tokenSet[.accessoryDetailButtonColor].dynamicColor)
         case .checkmark:
-            return UIColor(dynamicColor: fluentTheme.aliasTokens.brandColors[.primary])
+            return UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.brandForeground1])
```
Comment (huwilkes) [inline ios/FluentUI/Table View/TableViewCell.swift:47]: Do we know why this isn't in the table view cell tokens?

Comment (edjamesmsft) [reply ios/FluentUI/Table View/TableViewCell.swift:47]: I do not know why! I think that's outside the scope of this PR though. I can file a bug to track this.

File: ios/FluentUI.Demo/FluentUI.Demo/DemoColorThemes.swift [hunk 3]
```diff
@@ -22,82 +22,190 @@ enum DemoColorTheme: CaseIterable {
...
+        return UIColor(dynamicColor: DynamicColor(light: GlobalTokens.brandColors(.comm140),
+          dark: GlobalTokens.brandColors(.comm40)))
+    }
+
+    func brandStroke1(for theme: FluentUI.FluentTheme) -> UIColor? {
+        return UIColor(dynamicColor: DynamicColor(light: GlobalTokens.brandColors(.comm80),
+                                 dark: GlobalTokens.brandColors(.comm100)))
+    }
+
+    func brandStroke1Pressed(for theme: FluentUI.FluentTheme) -> UIColor? {
```
Comment (laminesm) [inline ios/FluentUI.Demo/FluentUI.Demo/DemoColorThemes.swift:114]: nit: Can this just be `FluentTheme`?

File: ios/FluentUI.Demo/FluentUI.Demo/DemoColorThemes.swift [hunk 4]
```diff
@@ -22,82 +22,190 @@ enum DemoColorTheme: CaseIterable {
         }
     }
 
-    var provider: ColorProviding? {
+    var provider2: ColorProviding2? {
```
Comment (laminesm) [inline ios/FluentUI.Demo/FluentUI.Demo/DemoColorThemes.swift:25]: Curious when we are appending 2 to everything? Shouldn't we just plug `ColorProviding2` and leave everything else as is naming-wise?

**Review by laminesm — COMMENTED** [review]
Do you know why the date picker test didn't fail on this PR?

File: ios/FluentUI.Demo/FluentUI.Demo/DemoColorThemes.swift (hunk 4 above)
Comment (edjamesmsft) [reply ios/FluentUI.Demo/FluentUI.Demo/DemoColorThemes.swift:25]: I will remove it! I had them side-by-side to keep things building and obvious while I was replacing the functionality, so it's a side effect of that.

**Comment by edjamesmsft** [comment]
> Do you know why the date picker test didn't fail on this PR?

I merged in your changes before I pushed the PR, so it is disabled :)

**Review by laminesm — APPROVED** [review]

**Pull request merged**